src/main/java/Doemon.java

```diff
@@ -0,0 +1,189 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Doemon {
+    /**
+     * Picture of Doemon made using text
+     */
+    private static String logo = "                       _______________\n" +
+            "                      /  --. --.      \\ \n" +
+            "                     /  | '| ' |   \\   \\ \n" +
+            "                    / /  `-O--'     \\   \\ \n" +
+            "                   |.  --  |  --     |   |\n" +
+            "                   |  --   |  --     |   |\n" +
+            "                    \\  (___|_______) /  /\n" +
+            "                     \\              /  /\n" +
+            "                       |== (t) ===|____";
+
+    /**
+     * Introduction string that is printed when Doemon is started.
+     */
+    private static String introStr = "Hello I'm\n" + logo + "\t\t\tDoemon!";
+
+    /**
+     * The string that is printed when Doemon is exited.
+     */
+    private static String exitStr = "I'm going to sleep now...See you again soon!";
+
+    /**
+     * A fixed-sized string array that stores user input.
+     */
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        // Introduction
+        System.out.println(output(introStr));
+
+        new Doemon().start();
+    }
+
+    /**
+     * Starts the Doemon chatbot.
+     */
+    private void start() {
+        // Handling inputs
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            String inputStr = sc.nextLine();
+
+            // Exit the chatbot
+            if (inputStr.equals("bye")) {
+                System.out.println(output(exitStr));
+                break;
+            }
+
+            // List tasks
+            if (inputStr.equals("list")) {
+                int listNum = 1;
+                String listStr = "Here is what's on my bread:\n\t";
+                for (Task task : tasks) {
+                    if (task == null) break;
+                    listStr += listNum++ + "." + task.toString() + "\n\t";
```

Comment:
Concatenating strings may not be very efficient. Maybe you can consider using StringBuilder instead? 

```
if (inputStr.equals("list")) {
    int listNum = 1;
    StringBuilder listStr = new StringBuilder("Here is what's on my bread:\n\t");
    for (Task task : tasks) {
        if (task == null) break;
        listStr.append(listNum++).append(".").append(task).append("\n\t");
    }
    listStr = new StringBuilder(listNum == 1 ? "You have no tasks!" : listStr.toString().trim());
    System.out.println(output(listStr.toString()));
    continue;
}
```